addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1258,6 +1262,94 @@ options.registry.parallax = options.Class.extend({
     onMove: function () {
         this._refreshPublicWidgets();
     },
+
+    //--------------------------------------------------------------------------
+    // Options
+    //--------------------------------------------------------------------------
+
+    /**
+     * Build/remove parallax.
+     *
+     * @see this.selectClass for parameters
+     */
+    parallax: function (previewMode, widgetValue, params) {
+        const isParallax = widgetValue === 'true';
+        this.$target.toggleClass('parallax', isParallax);
+        if (isParallax) {
+            this._buildParallax();
+        } else {
+            this._removeParallax();
+        }
+        if (!previewMode) {
+            this.isParallax = isParallax ? true : false;
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    _computeVisibility: function () {
+        let $el = this.$target.find('> .s_parallax_bg');
+        $el = $el[0] ? $el : this.$target;
+        const hasBackgroundImage = $el.css('background-image') !== 'none';
+        if (!hasBackgroundImage) {
+            this._removeParallax();
+            this.$target.removeClass('oe_img_bg');
+        }
+        const showParallaxOptions = hasBackgroundImage && !this.$target.hasClass('o_background_video');
+        return this._super(...arguments) && showParallaxOptions;
+    },
+    /**
+     * @private
+     */
+    _buildParallax: function () {
+        if (!this.$target.find('> .s_parallax_bg').length) {
+            $('<span/>', {
+                class: 's_parallax_bg oe_img_bg ',
+                style: 'background-position: 50% 0;',
+            }).prependTo(this.$target);
```

Comment:
Do you need to duplicate the creation of the parallax element? Since we need to keep the creation by the animation for compatibility, is this not enough?

Reply:
My thoughts were to add that to the options as needed and leave it in the animation for compatibility. but it's true that it's useless

Reply:
Debatable. I think the most consistent way would be to create an utility function to build that and call it in the option (for normal behaviour) and in the animation (for compatibility). But leaving the code in the animation only is good enough to me :+1: